cmake/CPM.cmake

```
@@ -309,6 +319,12 @@ function(CPMAddPackage)
     endif()
   endif()
 
+  # Check for required arguments
+
+  if(NOT DEFINED CPM_ARGS_NAME)
+    message(FATAL_ERROR "CPM: 'NAME' was not provided for package added with arguments: '${ARGN}'")
```

I see that this will become the predominant way of adding packages. Perhaps it may be good to explain here that `NAME` could not be inferred automatically, as it may confuse new users.

```suggestion
    message(FATAL_ERROR "CPM: 'NAME' was not provided and couldn't be automatically inferred for package added with arguments: '${ARGN}'")
```